src/qml/WiFiPage.qml

```diff
@@ -0,0 +1,149 @@
+ */
+
+import QtQuick 2.9
+import org.asteroid.controls 1.0
+import org.asteroid.utils 1.0
+import Connman 0.2
+import QtQuick.VirtualKeyboard 2.4
+
+Item {
+    id: root
+    TechnologyModel {
+        id: wifiModel
+        name: "wifi"
+        onCountChanged: {
+            console.log("COUNT CHANGE " + count)
+        }
+        onScanRequestFinished: {
+            console.log("SCAN FINISH")
+        }
+    }
+
+    NetworkTechnology {
+        id: wifiStatus
+        path: "/net/connman/technology/wifi"
+        onPoweredChanged: {
+            console.log("POWER CHANGE ") + powered
+            if (powered)
+                wifiModel.requestScan()
+        }
+    }
+
+    ListView {
+        id: wifiList
+        model: wifiModel
+        width: parent.width*0.7
+        anchors.horizontalCenter: parent.horizontalCenter
+        height: parent.height
+        header: Item {
+            //this is literally a statuspage
+            width: root.width
+            height: wifiStatus.powered ? width*0.6 : root.height
+            Behavior on height { NumberAnimation { duration: 100 } }
+            anchors.horizontalCenter: parent.horizontalCenter
+            Rectangle {
+                id: statusIconBackground
+                anchors.centerIn: parent
+                anchors.verticalCenterOffset: -parent.width*0.13
+                color: "black"
+                radius: width/2
+                opacity: wifiStatus.powered ? 0.4 : 0.2
+                width: parent.width*0.25
+                height: width
+                Icon {
+                    id: statusIcon
+                    anchors.fill: statusIconBackground
+                    anchors.margins: parent.width*0.12
+                    name: wifiStatus.powered ? "ios-wifi" : "ios-wifi-outline"
+                }
+                MouseArea {
+                    id: statusMA
+                    enabled: true
+                    anchors.fill: parent
+                    onClicked: wifiStatus.powered = !wifiStatus.powered
+                }
+            }
+
+
+            Label {
+                id: statusLabel
+                font.pixelSize: parent.width*0.07
+                horizontalAlignment: Text.AlignHCenter
+                verticalAlignment: Text.AlignVCenter
+                wrapMode: Text.Wrap
+                anchors.left: parent.left; anchors.right: parent.right
+                anchors.leftMargin: parent.width*0.04; anchors.rightMargin: anchors.leftMargin
+                anchors.verticalCenter: parent.verticalCenter
+                anchors.verticalCenterOffset: parent.width*0.15
+                text: "<h3>" + (wifiStatus.powered ? qsTrId("id-wifi-on"): qsTrId("id-wifi-off")) + "</h3>\n" + (wifiStatus.connected ? qsTrId("id-wifi-connected") : qsTrId("id-wifi-disconnected"))
+            }
+        }
+
+        footer: Item {height: wifiStatus.powered ? root.height*0.15 : 0; width: parent.width}
+
+        delegate: MouseArea {
```

Comment:
I wonder if we should use a https://github.com/AsteroidOS/qml-asteroid/blob/master/src/controls/qml/ListItem.qml here instead of re-inventing that wheel.

We can also have the ListItem use a Marquee, I think it's a sane default anyway.

Reply:
this does show two bits of text instead of one, and doesn't make use of an icon, so ListItem isn't relevant here

Reply:
qml-asteroid is not set in stone. We can expand ListItem to handle a primary and a secondary text and no icon.

The reason why I think this is relevant is that this sort of code factorization also keeps the design consistent across the OS. For example, ListItem has a HighlightBar that isn't used here. The more we factorize code the better it is to keep the UI sharp and to conduct refactorings/redesigns in the future.

Reply:
The highlightbar is something I wanted to add, but I wanted to push the code to prompt the security conversation before I invested too much time. 
I want to keep this code here without abstracting it away, but I'm open to the conversation. However:
- while adding components to org.asteroid.controls can improve consistency, it won't necessarily improve maintainability. Recent changes which added highlightbar immediately broke asteroid-settings, but that was fixed - however, asteroid-map still hasn't, and the new highlightbar changes have broken the UI significantly. I think future changes, apart from miniscule tweaks to font size and the like, will probably bring the same kind of breakage and the same kind of cross-repo maintenance tasks. 
- qml-asteroid won't account for all the edge cases, and I think this might be an edge case. This chunk of code definitely needs cleaning up and making more consistent, but I'm not sure it will be used anywhere else, and I think 'potential to be used somewhere else' is the bar we should set for adding things to qml-asteroid.